File: Searching Algorithms/Java/LinearSearch.java
public class LinearSearch {

    private static int linearSearch(int array[], int key){
        for(int i=0; i< array.length; i++){
            if(array[i] == key){
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int array[] = {8,7,2,3,4,1};
        int key = 4;

        int result = linearSearch(array, key);
        if(result == -1){
            System.out.println("Element is not present in array");
        } else {
            System.out.println("Element is present ar index " + result);
        }
    }
}

File: Searching Algorithms/C++/LinearSearch.cpp
#include <iostream>
using std::cout;
using std::endl;
using std::vector;

//Time O(n) | Space O(1)
int linearSearch(vector<int> vector, int key){
    for(int i=0; i<vector.size(); i++){
        if(vector[i] == key){
            return i;
        }
    }
    return  -1;
}

int main(int argc, char const *argv[])
{
    vector<int> input = {2,3,1,5,6,10};
    int key = 5;
    int result = linearSearch(input, key);
    
    if(result == -1){
        cout << "Element is not present in vector" << endl;
    } else {
        cout << "Element is present at index " << result <<endl;
    }
    return 0;
}
